File: CoreAccess.Models/ApplicationModel.cs
namespace CoreAccess.Models;

public class ApplicationSearchOptions
{
    public string Search { get; set; }
    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 10;
}

public class ApplicationUpdateRequest
{
    public string? ClientId { get; set; }
    public string? DisplayName { get; set; }
    public IEnumerable<string>? RedirectUris { get; set; }
    public IEnumerable<string>? PostLogoutRedirectUris { get; set; }
    public string? ClientSecret { get; set; }
    public IEnumerable<string>? Permissions { get; set; }
    public IEnumerable<string>? Requirements { get; set; }
}

public class ApplicationDto
{
    public string Id { get; set; }
    public string ClientId { get; set; }
    public string DisplayName { get; set; }
    public string ClientType { get; set; }

}

public class ApplicationDetailDto : ApplicationDto
{
    public IEnumerable<string>? RedirectUris { get; set; }
    public IEnumerable<string>? PostLogoutRedirectUris { get; set; }
    public string ApplicationType { get; set; }
    public string ConsentType { get; set; }
    public string ClientSecret { get; set; }
    public IEnumerable<string>? Permissions { get; set; }
    public string? Requirements { get; set; }
}

File: CoreAccess.Frontend/coreaccess-webui/src/model/ApplicationModel.ts
export interface ApplicationSearchOptions {
    search: string;
    page?: number;        // Default in C# war 1
    pageSize?: number;    // Default in C# war 10
}

export interface ApplicationUpdateRequest {
    clientId?: string;
    displayName?: string;
    redirectUris?: string[];
    postLogoutRedirectUris?: string[];
    clientSecret?: string;
    permissions?: string[];
    requirements?: string[];
}

export interface ApplicationDto {
    id: string;
    clientId: string;
    displayName: string;
    clientType: string;
}

export interface ApplicationDetailDto extends ApplicationDto {
    redirectUris?: string[];
    postLogoutRedirectUris?: string[];
    applicationType: string;
    consentType: string;
    clientSecret: string;
    permissions?: string[];
    requirements?: string;
}
